src/integrations/watchers/indexable-permalink-watcher.php

```diff
@@ -144,7 +166,7 @@ public function reset_permalinks_term( $old, $new, $type ) {
 	 * @return bool Whether the reset request ran.
 	 */
 	public function force_reset_permalinks() {
-		if ( $this->should_reset_permalinks() ) {
+		if ( $this->get_reset_reason() !== null ) {
```

Comment:
As far as I see, `get_reset_reason()` is called only when `reset_permalinks()` runs so at this point is always null because `reset_permalinks()`  didn't run yet? Totally possible I'm missing something though.